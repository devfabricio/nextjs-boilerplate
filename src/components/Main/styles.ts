import styled from 'styled-components'

export const Wrapper = styled.main`
  align-items: center;
  background-color: #222c59;
  color: #fff;
  display: flex;
  flex-direction: column;
  width: 100%;
  height: 100%;
  justify-content: center;
  padding: 3rem;
  text-align: center;
`

export const Logo = styled.img`
  margin-bottom: 3rem;
  width: 40rem;
`

export const Description = styled.h2`
  font-size: 2.5rem;
  font-weight: 400;
  line-height: 4rem;
  max-width: 60rem;
`
