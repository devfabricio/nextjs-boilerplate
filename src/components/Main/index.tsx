import * as S from './styles'

const Main = () => (
  <S.Wrapper>
    <S.Logo src={'/img/logo-dark.png'} alt={'Logo da Hello Codes'} />
    <S.Description>
      A <b>NextJS</b> boilerplate with Typescript, Babel, Storybook, Jest, React
      Testing Library, Eslint, Husky, Lint Staged and Plop.
    </S.Description>
  </S.Wrapper>
)

export default Main
