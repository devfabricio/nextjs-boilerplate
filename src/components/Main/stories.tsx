import Main from './index'
import { Meta, Story } from '@storybook/react'

export default {
  title: 'Main',
  component: Main
} as Meta<typeof Main>

export const Basic: Story = () => <Main />
