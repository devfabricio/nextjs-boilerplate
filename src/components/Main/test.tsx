import { render } from '@testing-library/react'

import Main from '.'

describe('<Main/>', () => {
  it('should render the colors correctly', function () {
    const { container } = render(<Main />)
    expect(container.firstChild).toHaveStyle({ 'background-color': '#222c59' })
    expect(container.firstChild).toMatchSnapshot()
  })
})
