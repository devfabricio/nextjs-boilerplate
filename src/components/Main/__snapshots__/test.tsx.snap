// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Main/> should render the colors correctly 1`] = `
.c0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #222c59;
  color: #fff;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  width: 100%;
  height: 100%;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 3rem;
  text-align: center;
}

.c1 {
  margin-bottom: 3rem;
  width: 40rem;
}

.c2 {
  font-size: 2.5rem;
  font-weight: 400;
  line-height: 4rem;
  max-width: 60rem;
}

<main
  class="c0"
>
  <img
    alt="Logo da Hello Codes"
    class="c1"
    src="/img/logo-dark.png"
  />
  <h2
    class="c2"
  >
    A 
    <b>
      NextJS
    </b>
     boilerplate with Typescript, Babel, Storybook, Jest, React Testing Library, Eslint, Husky, Lint Staged and Plop.
  </h2>
</main>
`;
